### DISCUSSION TITLE
Using QT file picker instead of GTK? Does zim support gtk_use_portal=1?

### DISCUSSION THREAD

**Discussion by thatguychuck** [zim-desktop-wiki/zim-desktop-wiki#1696] [Q&A]
I'm on plasma-desktop and have many favorites and such added to dolphin. My GTK file picker is rather empty and makes attaching files a little cumbersome. I have the following packages installed:
`xdg-desktop portal xdg-desktop-portal-kde`

I've been playing around with the GTK_USE_PORTAL=1 but it seems to be ignored. Is this not supported? Or am I going about it the wrong way. Has anyone else been able to use dolphin as the default file picker?

**Comment by jaap-karssenberg** [comment]
I would refer to generic documentation on this "portal" configuration.
Never heard of it and does not sound as something that is zim specific. Do
you have other Gtk applications that work with this feature?

-- Jaap

**Comment by thatguychuck** [comment]
I remember having to do some fiddling with Firefox to get this going. I don't have the exact guide I used but a good representation is here: [https://askubuntu.com/questions/1100261/how-do-i-make-firefox-64-use-the-kde-file-selection-dialog](https://askubuntu.com/questions/1100261/how-do-i-make-firefox-64-use-the-kde-file-selection-dialog)

From what I'm reading, it's part of the gnome took kit and gets set application side. [https://docs.gtk.org/gtk3/class.FileChooserNative.html](https://docs.gtk.org/gtk3/class.FileChooserNative.html)

Unfortunately my coding skills are nil so I'm not sure I can come up with a patch, IF it's even something you are interested in. From my cursory googling it sounds like going this route may introduce new bugs and the file chooser might not behave as expected. At any rate I believe I found a workaround, which is posted below for anyone else that may find their way here.

All in all this may have ended up being an XY problem. I don't believe there are any "features" in the Qt file chooser that I need, other than my bookmarks.  I found a recommendation on the subject in the [arch wiki](https://wiki.archlinux.org/title/Uniform_look_for_Qt_and_GTK_applications#QGtkStyle) to sync my dolphin bookmarks  to the gtk side using the following command:
`awk -F\" '/<bookmark href=\"file/ {print $2}' < $HOME/.local/share/user-places.xbel > $HOME/.config/gtk-3.0/bookmarks`
Note: On my system this produced a warning about a regexp escape sequence, but it still worked as intended.

There is even an application out there that manages bookmarks between GTK+ and KDE. [https://github.com/jlu5/bookmarksync](https://github.com/jlu5/bookmarksync)

This becomes a manual thing, needed to be re run as my bookmarks change, but produces the desired effect in Zim. Using this command also has the added benefit of being universal to any other gtk programs I may run with that file chooser.

**Comment by jaap-karssenberg** [comment]
Ah OK, now I understand how this works. Yes would require a patch to use
this FileChooserNative instead of the default Gtk file chooser. The
downside of this is that some functionality will be lost. E.g. secondary
actions / checkboxes in the file dialogs. This may not be problematic, but
will require having this option switched by a preference. (Or switched by
an environment parameter, which might help integrating with flatpack ?)

Regards,

Jaap
